modules/andes-core/broker/src/main/java/org/wso2/andes/amqp/QpidAndesBridge.java

```diff
@@ -466,7 +466,8 @@ public static void removeBinding(Binding binding) throws AndesException {
      * @param subscription subscription
      * @throws AndesException
      */
-    private static void addLocalSubscriptionsForAllBindingsOfQueue(AMQQueue queue, Subscription subscription) throws AndesException, SubscriptionAlreadyExistsException {
+    private static void addLocalSubscriptionsForAllBindingsOfQueue(AMQQueue queue,
+                                                                   Subscription subscription) throws SubscriptionAlreadyExistsException, AMQException, AndesException {
```

Comment:
We still throw AndesException because it throws another andes exception in this method. However this is a private method. This AndesException is caught  in the method where it is called which is also inside this class and throws AMQ exception. We do some stuff in the catch clause and that also throws andes exception. That is why it is handle that way.  HasithaH please correct me if I'm wrong